src/waveform/renderers/allshader/waveformrendererpreroll.cpp

```diff
@@ -6,13 +6,13 @@
 #include <array>
 
 #include "skin/legacy/skincontext.h"
-#include "util/texture.h"
 #include "waveform/renderers/allshader/matrixforwidgetgeometry.h"
 #include "waveform/renderers/waveformwidgetrenderer.h"
 #include "widget/wskincolor.h"
 
 namespace {
-std::unique_ptr<QOpenGLTexture> generateTexture(float markerLength,
+void generateTexture(OpenGLTexture2D* pTexture,
```

Comment:
I really don't see how the tradeoff in code quality makes sense. There is no reason to introduce an out-parameter here IMO. If the extra allocation bothers you, why not return it by value directly?

Reply:
This is this no out parameter which may be considered as bad code quality these days. It just updates the texture via pointer. I kept just to avoid moving code around. I will now make it a normal member function as usual and rename it to avoid future confusion.

Reply:
Refactored to become drawPrerollImage() returning the temporary QImage.